vit-servicing-station-lib/src/db/models/proposals.rs

```diff
@@ -109,46 +113,55 @@ pub struct Proposal {
     pub fund_id: i32,
     #[serde(alias = "challengeId")]
     pub challenge_id: i32,
-    #[serde(
-        alias = "proposalSolution",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_solution: Option<String>,
-    #[serde(
-        alias = "proposalBrief",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_brief: Option<String>,
-    #[serde(
-        alias = "proposalImportance",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_importance: Option<String>,
-    #[serde(
-        alias = "proposalGoal",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_goal: Option<String>,
-    #[serde(
-        alias = "proposalMetrics",
-        default,
-        skip_serializing_if = "Option::is_none"
-    )]
-    pub proposal_metrics: Option<String>,
+}
+
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub enum ProposalChallengeInfo {
+    Simple(simple::ChallengeInfo),
+    CommunityChoice(community_choice::ChallengeInfo),
+}
+
+#[derive(Serialize, Deserialize)]
+struct SerdeProposalChallengeInfo {
+    #[serde(flatten, default, skip_serializing_if = "Option::is_none")]
+    simple: Option<simple::ChallengeInfo>,
+    #[serde(flatten, default, skip_serializing_if = "Option::is_none")]
+    community: Option<community_choice::ChallengeInfo>,
 }
 
 #[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone)]
 pub struct FullProposalInfo {
     #[serde(flatten)]
     pub proposal: Proposal,
+    #[serde(flatten)]
+    pub challenge_info: ProposalChallengeInfo,
     #[serde(alias = "challengeType")]
     pub challenge_type: ChallengeType,
 }
 
+impl Serialize for ProposalChallengeInfo {
+    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+    where
+        S: Serializer,
+    {
+        let serde_data: SerdeProposalChallengeInfo = self.clone().into();
```

Comment:
There is some proxy type form of `serde` attribute to avoid typing this explicitly, I think.

Reply:
`serde(remote)`, but in this case, it wouldn't work because it may fail to transform, we cannot implement a `From` safely for this.